contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -3090,6 +3127,45 @@ void removeCtxStringFromQuery(PLtsql_expr* expr, ParserRuleContext *ctx, ParserR
 	replaceTokenStringFromQuery(expr, ctx->getStart(), ctx->getStop(), NULL, baseCtx);
 }
 
+void extractTableHintsFromOptionClause(TSqlParser::Option_clauseContext *octx) {
+	for (auto option: octx->option()) {
+		if(option->TABLE()) {
+			std::string table_name = ::getFullText(option->table_name());
+			for(auto table_hint: option->table_hint()) {
+				extractTableHint(table_hint, table_name);
+			}
+		}
+	}
+}
+
+void extractTableHints(TSqlParser::With_table_hintsContext *tctx, std::string table_name) {
+	for (auto table_hint: tctx->table_hint())
+		extractTableHint(table_hint, table_name);
+}
+
+
+void extractTableHint(TSqlParser::Table_hintContext *table_hint, std::string table_name) {
```

Comment:
If there is no written coding convention, then it would be better to follow conventions in the same file. Most of the code in this file look like the following:
```
void extractTableHint(...)
{
    if(table_hint->INDEX())
    {
        ...
    }
}
```